tests/unit/expiry/HgvTrlFirstTestStrategy.unitTest.ts

```diff
@@ -0,0 +1,90 @@
+import { cloneDeep } from "lodash";
+import testResults from "../../resources/test-results.json";
+import { HgvTrlFirstTestStrategy } from "../../../src/handlers/expiry/strategies/HgvTrlFirstTestStrategy";
+import { TestTypeForExpiry } from "../../../src/models/TestTypeforExpiry";
+import { DateProvider } from "../../../src/handlers/expiry/providers/DateProvider";
+import { TestType } from "../../../src/models/ITestResult";
+import { VEHICLE_TYPE } from "../../../src/assets/Enums";
+
+describe("HgvTrlFirstTestStrategy", () => {
+  let testResultsMockDB: any;
+  let hgvTrlFirstTestStrategy: HgvTrlFirstTestStrategy;
+
+  beforeEach(() => {
+    testResultsMockDB = cloneDeep(testResults);
+  });
+
+  afterEach(() => {
+    // reset date to current date
+    hgvTrlFirstTestStrategy.dateProvider.setTestDate(new Date());
+  });
+
+  context("for hgv vehicle type", () => {
+    describe("with a valid registration date", () => {
+      describe(" and the test is conducted exactly 2 month from registration anniversary", () => {
```

Comment:
There are 4 scenarios given on https://wiki.dvsacloud.uk/display/HVT/Expiry+Date+Logic+-+HGV  but only a couple of unit tests here.  Where are the others? Also, we should be testing boundaries and leap years etc. rather than a single case.

Using the jest tables to input the scenario's would make this much easier to read, make the test code more concise (it's very WET) as well as making it easy to get the business to sign off on the test cases

Reply:
All existing tests updated according to the jest table format and strategy is mocked to make the code DRY.